File: java/src/test/java/problems1_100/WordSearchTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class WordSearchTest extends SampledTest {
    @Test
    public void test() {
        for (WordSearch solution : WordSearch.values()) {
            for (WordSearchSample sample : getSamples(WordSearchSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.exist(sample.input.board, sample.input.word))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/WordSearchTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class WordSearchTest : SampledTest() {
    @Test
    fun test() =
        WordSearch.entries.forEach { solution ->
            getSamples<Array<WordSearchSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.exist(input.board, input.word))
                    .isEqualTo(output)
            }
        }
}
